Lib/test/test_threading.py

```diff
@@ -1338,6 +1338,21 @@ def run(self):
         # explicitly break the reference cycle to not leak a dangling thread
         thread.exc = None
 
+    def test_multithread_modify_file_noerror(self):
+        # See issue25872
+        def modify_file():
+            with open(os_helper.TESTFN, 'w', encoding='utf-8') as fp:
+                fp.write(' ')
```

Comment:
If you don't call traceback.format_stack() then this test doesn't check for the bug in the issue.

Make sure the test fails in the current version of the code.

Reply:
I have tried to run python 3.7.3 and 3.8.1 (which before the following patch) on the container , but I could not reproduce KeyError.
https://github.com/python/cpython/pull/18007

So I have reverted to using traceback as reported. I apologize for the confusion.